File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        bucket = [[] for _ in range(len(nums) + 1)]

        d = {}
        for num in nums:
            d[num] = d.get(num, 0) + 1
        
        for key, value in d.items():
            bucket[value].append(key)
        
        res = []
        for i in range(len(bucket) - 1, 0, -1):
            for num in bucket[i]:
                res.append(num)
                k -= 1
                if k == 0: return res
        return res

File: Data Structures & Algorithms/majority-element/submission-3.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 0, ans = 0;
        for (int num: nums){
            if (count == 0){
                ans = num;
            }
            count += (num == ans) ? 1 : -1;
        }
        return ans;
    }
}